src/tbb/arena.cpp

```diff
@@ -90,6 +91,14 @@ template <bool as_worker>
 std::size_t arena::occupy_free_slot(thread_data& tls) {
     // Firstly, external threads try to occupy reserved slots
     std::size_t index = as_worker ? out_of_arena : occupy_free_slot_in_range( tls,  0, my_num_reserved_slots );
+    if (index == out_of_arena && has_numa_optimization()) {
+        std::pair<int, int> my_range = {my_num_reserved_slots, info::default_concurrency()};
+        auto it = numa_intervals.find(governor::get_my_current_numa_node());
+        if (it != numa_intervals.end())
+            my_range = it->second;
+        index = occupy_free_slot_in_range(tls, max(static_cast<int>(my_num_reserved_slots), my_range.first),
```

Comment:
But my_num_reserved_slots always equal to my_range.first, because you init my_range.first with this value